### PR TITLE
Update to new NetQASM interface

### LINKED ISSUE #6: Update to be compatible with new NetQASM interface/features
Make SquidASM compatible again to NetQASM after https://github.com/QuTech-Delft/netqasm/pull/14

### PR DESCRIPTION
**Pull request by bvdvecht**
Closes #6

### CODE REVIEW

**Comment by bvdvecht** [comment]
Blocked until https://github.com/QuTech-Delft/netqasm/pull/14 is merged.

**Comment by bvdvecht** [comment]
Unblocked now that https://github.com/QuTech-Delft/netqasm/pull/14 has been merged.

File: squidasm/sim/stack/processor.py [hunk 1]
```diff
@@ -586,17 +608,6 @@ def _interpret_two_qubit_instr(
 
 
 class GenericProcessor(Processor):
-    def _interpret_qalloc(self, app_id: int, instr: core.QAllocInstruction) -> None:
```
Comment (QFer) [inline squidasm/sim/stack/processor.py:589]: I don't get the move to the (abstract) base class for this method. The NVProcessor has its own implementation, so you could have kept it an abstract method (in the sense no implementation in the abstract base class Processor) and another specific implementation in GenericProcessor.

File: examples/stack/bqc/example_bqc_nv.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        conn.insert_breakpoint(BreakpointAction.DUMP_LOCAL_STATE)
+        m2 = epr1.measure(store_array=False)
+        yield from conn.flush()
+
+        m2 = int(m2)
+        return {"m1": m1, "m2": m2}
+
+
+PI = math.pi
+PI_OVER_2 = math.pi / 2
```
Comment (QFer) [inline examples/stack/bqc/example_bqc_nv.py:163]: You just defined math.pi as PI: why not: PI / 2

File: examples/stack/bqc/example_bqc_nv.py [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+
+    frac_fail = round(num_fails / num_times, 2)
+    print(f"fail rate: {frac_fail}")
+
+
+if __name__ == "__main__":
+    num_times = 100
+    LogManager.set_log_level("WARNING")
+
+    # num_times = 1
```
Comment (QFer) [inline examples/stack/bqc/example_bqc_nv.py]: better delete these 2 lines

**Review by QFer — CHANGES_REQUESTED** [review]

File: squidasm/sim/stack/processor.py (hunk 1 above)
Comment (bvdvecht) [reply squidasm/sim/stack/processor.py:589]: The implementation that is now in the base Processor class is expected to be re-used by new derived classes of Processor in the future. So in a way this is preparing for adding new Processor subclasses and not having to add the same implementation of `_interpret_qalloc` in those.

File: examples/stack/bqc/example_bqc_nv.py (hunk 2 above)
Comment (bvdvecht) [reply examples/stack/bqc/example_bqc_nv.py:163]: I think this is mostly a difference in taste. Or do you have a concrete reason to prefer one way of writing over the other?

File: examples/stack/bqc/example_bqc_nv.py (hunk 3 above)
Comment (bvdvecht) [reply examples/stack/bqc/example_bqc_nv.py]: Done

**Review by QFer — APPROVED** [review]

**Pull request merged**